flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -79,20 +86,45 @@ public void testPerJobMode() throws Exception {
 
 			env.addSource(new InfiniteSource())
 				.shuffle()
-				.addSink(new DiscardingSink<Integer>());
+				.addSink(new DiscardingSink<>());
 
 			final JobGraph jobGraph = env.getStreamGraph().getJobGraph();
 
 			File testingJar = YarnTestBase.findFile("..", new TestingYarnClusterDescriptor.TestJarFinder("flink-yarn-tests"));
```

Comment:
`TestingYarnClusterDescriptor` is inherited from `LegacyYarnClusterDescriptor` and should be removed. Here we rely on `TestingYarnClusterDescriptor.TestJarFinder`, which is a util class, and could move it into `YarnTestUtils` to release such dependency.

Reply:
Good idea. I'll move `TestJarFinder` into `YarnTestUtils`. However, we cannot remove `TestingYarnClusterDescriptor` at the moment, because it is used by the `YARNHighAvailabilityITCase`.